pkg/sink/initialization.go

```diff
@@ -166,12 +168,36 @@ func ConfigureClients(ctx context.Context, clusterConfig *rest.Config) (Clients,
 		return Clients{}, xerrors.Errorf("Failed to create TriggersClient: %s", err)
 	}
 	ceClient := cloudevent.Get(ctx)
-
+	certData, err := getCertificateData(ctx, kubeClient)
+	if err != nil {
+		return Clients{}, xerrors.Errorf("Failed to get certs from secret: %s", err)
+	}
 	return Clients{
 		DiscoveryClient: kubeClient.Discovery(),
 		RESTClient:      kubeClient.RESTClient(),
 		TriggersClient:  triggersClient,
 		K8sClient:       kubeClient,
 		CEClient:        ceClient,
+		CertData:        certData,
 	}, nil
 }
+
+// getCertificateData is used to get certData which is used by eventlistener while connecting to clusterinterceptor
+func getCertificateData(ctx context.Context, kubeClient *kubeclientset.Clientset) ([]byte, error) {
+	var certData []byte
+	allSecrets, err := kubeClient.CoreV1().Secrets("").List(ctx, metav1.ListOptions{})
```

Comment:
why are we listing all secrets instead of just a get call given that we already know the name?

Reply:
also the fact that the EL has to know the exact secret name for the core interceptors seems not ideal. Ideally, the EL should be able to pick up where to pick up the certData from the CRD itself (e.g. the webhooks contain a caCert field with this info)